packages/bp/src/core/dialog/janitor.ts

```diff
@@ -93,31 +97,34 @@ export class DialogJanitor extends Janitor {
       const { channel, target, threadId } = SessionIdFactory.extractDestinationFromId(sessionId)
       const session = await this.sessionRepo.get(sessionId)
 
-      this.dialogEngine.detectInfiniteLoop(session.context.jumpPoints || [], botId)
+      await this.detectTimeoutInfiniteLoop(session, botId)
 
-      // This event only exists so that processTimeout can call processEvent
-      const fakeEvent = Event({
+      // Used to call processEvent and it's also saved for loop detection
+      const InternalEvent = Event({
         type: 'timeout',
         channel,
         target,
         threadId,
-        direction: 'incoming',
+        direction: 'internal',
```

Comment:
direction internal ? not sure we should introduce a new direction

Reply:
We need to store this event in order to detect loops; I don't think that its purpose fits either as an 'incoming' or 'outgoing' event; my idea was to make it more explicit; I can make a lot of tests (I already did, but I will do it again) to be double sure that every place that direction is used will not break anything if there is a timeout (Before, we were not storing this incoming event, In my opinion, storing an invalid incoming event will have more consequences than creating a new direction, so this type of event can be filtered), deal? @EFF